### PR TITLE
 Fix nerdctl load stdin for pipes.

### PR DESCRIPTION
**Pull request by manugupt1**
Fixes https://github.com/containerd/nerdctl/issues/1404

### CODE REVIEW

File: cmd/nerdctl/load_linux_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+	f, err := os.OpenFile(filepath.Join(tmp, "load.sh"), os.O_CREATE|os.O_RDWR, 0777)
+	assert.NilError(t, err)
+
+	f.WriteString("#/bin/sh\n")
+	f.WriteString(fmt.Sprintf("cat %s/common.tar | %s", tmp, cmdStr))
+	err = f.Close()
+	assert.NilError(t, err)
+
+	combined, err := exec.Command("/bin/sh", filepath.Join(tmp, "load.sh")).CombinedOutput()
```
Comment (AkihiroSuda) [inline cmd/nerdctl/load_linux_test.go:51]: Can we just use `sh -euxc`?

File: cmd/nerdctl/load_linux_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+	cmdStr := ""
+	for _, c := range base.Cmd("load").Command {
+		cmdStr += c + " "
+	}
+
+	f, err := os.OpenFile(filepath.Join(tmp, "load.sh"), os.O_CREATE|os.O_RDWR, 0777)
+	assert.NilError(t, err)
+
+	f.WriteString("#/bin/sh\n")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/load_linux_test.go:46]: Need `set -eu`

Comment (AkihiroSuda) [reply cmd/nerdctl/load_linux_test.go:46]: Also `set -eu -o pipefail` (with /bin/bash)

File: cmd/nerdctl/load_linux_test.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	"github.com/containerd/nerdctl/pkg/testutil"
+	"gotest.tools/v3/assert"
+)
+
+func TestLoadStdinFromPipe(t *testing.T) {
+	t.Parallel()
+	base := testutil.NewBase(t)
+
+	tmp := t.TempDir()
+	base.Cmd("save", testutil.CommonImage, "-o", filepath.Join(tmp, "common.tar")).AssertOK()
```
Comment (fahedouch) [inline cmd/nerdctl/load_linux_test.go:36]: How can you save an image before fetching it ?

Comment (manugupt1) [reply cmd/nerdctl/load_linux_test.go:37]: Other tests have pulled it; added an extra line.

File: cmd/nerdctl/load.go [hunk 4]
```diff
@@ -132,7 +132,7 @@ func loadImage(in io.Reader, cmd *cobra.Command, platMC platforms.MatchComparer,
 		if quiet {
 			fmt.Fprintln(cmd.OutOrStdout(), img.Target.Digest)
 		} else {
-			fmt.Fprintf(cmd.OutOrStdout(), "done\n")
+			fmt.Fprintf(cmd.OutOrStdout(), "Loaded image: %s", img.Name)
```
Comment (manugupt1) [inline cmd/nerdctl/load.go:135]: Makes the test docker compatible.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**